sonar-php-plugin/src/main/java/org/sonar/plugins/php/reports/phpstan/PhpStanJsonReportReader.java

```diff
@@ -47,7 +47,12 @@ static void read(InputStream in, Consumer<Issue> consumer) throws IOException, P
 
   private void read(InputStream in) throws IOException, ParseException {
     JSONObject rootObject = (JSONObject) jsonParser.parse(new InputStreamReader(in, UTF_8));
-    JSONObject files = (JSONObject) rootObject.get("files");
+    // SONARPHP-1316 : in case of no issue in report, 'files' in a JSONArray and not a JSONObject
+    JSONObject files = null;
+    Object filesObject = rootObject.get("files");
+    if (filesObject instanceof JSONObject) {
+      files = (JSONObject) filesObject;
+    }
     if (files != null) {
       files.forEach((file, records) -> onFile(cleanFilePath((String) file), (JSONObject) records));
     }
```

Comment:
```suggestion
    Optional.of(rootObject.get("files"))
      .filter(JSONObject.class::isInstance).map(JSONObject.class::cast)
      .ifPresent(files -> 
        files.forEach((file, records) -> onFile(cleanFilePath((String) file), (JSONObject) records)));
```
You can decide if it's more readable for you.